### PR TITLE
Remove unused classes  (jsp tools) from the JSP Impl

### PR DESCRIPTION
**Pull request by volosied**
It looks like JSP was picked  from WebSphere  and moved to Liberty. However, not all of the functionality (ie. Batch Compiler) was moved over. This meant that there are classes which are not used and unnecessarily supported.  

Packages removed:
## com.ibm.wsspi.jsp.tools
-  Private Package
- Not actually used as an SPI. (Only  `com.ibm.wsspi.jsp.taglib.config` is exported as an SPI) 
  - https://github.com/OpenLiberty/open-liberty/blob/7822343331e22b0b00df081916794b233e7f4247/dev/com.ibm.websphere.appserver.spi.jsp/bnd.bnd#L20-L22 
## com.ibm.ws.jsp.tools
- Private Package
- Slight refactoring was needed JspTranslatorUtil due to the the use of InnerclassFilenameFilter. 
- The JspToolsFactoryHelper implementation should load `JspToolsFactoryImpl` but this class does not exist. 
- CMVC also references `JspToolsImpl`, but I don't see this class either in Liberty. 

There are other packages that could also be removed, but they'll need a bit more investigation. 

As mentioned in a later comment, these jsp tools classes are used by the Admin App Mgmt to precompile the classes. This admin tool does not exist in OL, so these classes can be removed safely.

### CODE REVIEW

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat,io.openliberty.pages.3.1.internal_fat,io.openliberty.pages.4.0.internal_fat,com.ibm.ws.jsp_fat_lWAS,com.ibm.ws.jsp_fat_lWAS_jsp.2.3,com.ibm.ws.jsp_fat,com.ibm.ws.jsp_bvt,com.ibm.ws.jsp_fat_lWAS.toleration.ee9,com.ibm.ws.jsp_fat_lWAS.toleration.ee10,com.ibm.ws.jsf_fat,com.ibm.ws.jsf_fat_jsf2.2,com.ibm.ws.jsf_fat_jsf2.3,com.ibm.ws.jsf_fat_lWAS,com.ibm.ws.jsf_fat_lWAS_jsf2.2,com.ibm.ws.jsf_fat_lWAS_servlet3.1,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat

**Comment by volosied** [comment]
Everything passed except for one invalid bucket which I mistakenly specified - `Invalid test bucket - com.ibm.ws.jsp_bvt`

File: dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/utils/JspTranslatorUtil.java [hunk 1]
```diff
@@ -38,7 +36,7 @@
 import com.ibm.ws.jsp.configuration.JspConfiguration;
 import com.ibm.ws.jsp.taglib.TagLibraryCache;
 import com.ibm.ws.jsp.taglib.TagLibraryInfoImpl;
-import com.ibm.ws.jsp.tools.JspFileUtils.InnerclassFilenameFilter; //PI12939
+// import com.ibm.ws.jsp.tools.JspFileUtils.InnerclassFilenameFilter; //PI12939
```
Comment (volosied) [inline dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/utils/JspTranslatorUtil.java]: I could remove this rather than commenting it out

**Comment by volosied** [comment]
Can't access CMVC, but a search for `JspToolsImpl` found: https://www.ibm.com/support/pages/apar/PI61073

Looks like the JSP tools was used by the admin console to configure JSP options when deploying applications 

Another APAR - https://www.ibm.com/support/pages/apar/PI52135

I also see that the CompileJspTask.java class  (from app management) contains the relevant JSP admin code.  It references the jsp tools.

i.e:
   private static final String CLASS_NAME_JSP_TOOLS_FACTORY_HELPER =
      "com.ibm.wsspi.jsp.tools.JspToolsFactoryHelper";

**Comment by volosied** [comment]
Another APAR - https://www.ibm.com/support/pages/apar/PI52135

I can't find the admin code in Websphere, but the surrounding evidences points to the admin console.

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat,io.openliberty.pages.3.1.internal_fat,io.openliberty.pages.4.0.internal_fat,com.ibm.ws.jsp_fat_lWAS,com.ibm.ws.jsp_fat_lWAS_jsp.2.3,com.ibm.ws.jsp_fat,com.ibm.ws.jsp_fat_lWAS.toleration.ee9,com.ibm.ws.jsp_fat_lWAS.toleration.ee10,com.ibm.ws.jsf_fat,com.ibm.ws.jsf_fat_jsf2.2,com.ibm.ws.jsf_fat_jsf2.3,com.ibm.ws.jsf_fat_lWAS,com.ibm.ws.jsf_fat_lWAS_jsf2.2,com.ibm.ws.jsf_fat_lWAS_servlet3.1,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat

**Review by epj — APPROVED** [review]
looks OK

**Comment by volosied** [comment]
#libby

**Pull request merged**